File: src/network/packet/impl/ServerGroupSyncPacket.php
<?php

namespace pocketcloud\cloud\network\packet\impl;

use pocketcloud\cloud\group\ServerGroup;
use pocketcloud\cloud\network\client\ServerClient;
use pocketcloud\cloud\network\packet\ClientboundPacket;
use pocketcloud\cloud\network\packet\CloudPacket;
use pocketcloud\cloud\network\packet\util\PacketData;

final class ServerGroupSyncPacket extends CloudPacket implements ClientboundPacket {

    public function __construct(
        private readonly ?ServerGroup $group = null,
        private readonly bool $removal = false
    ) {}

    public function handle(ServerClient $client): void {}

    public function encodePayload(PacketData $packetData): void {
        $packetData->writeAll($this->group, $this->removal);
    }

    public function decodePayload(PacketData $packetData): void {}

    public function getGroup(): ?ServerGroup {
        return $this->group;
    }

    public function isRemoval(): bool {
        return $this->removal;
    }

    public static function create(ServerGroup $group, bool $removal): self {
        return new self($group, $removal);
    }
}

File: src/main/java/de/pocketcloud/cloud/network/packet/impl/ServerGroupSyncPacket.java
package de.pocketcloud.cloud.network.packet.impl;

import de.pocketcloud.cloud.template.group.ServerGroup;
import de.pocketcloud.cloud.network.client.ServerClient;
import de.pocketcloud.cloud.network.packet.ClientboundPacket;
import de.pocketcloud.cloud.network.packet.CloudPacket;
import de.pocketcloud.cloud.network.packet.data.PacketData;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.jetbrains.annotations.NotNull;

@NoArgsConstructor
@Getter
public final class ServerGroupSyncPacket extends CloudPacket implements ClientboundPacket {

    private ServerGroup group;
    private boolean removal;

    public ServerGroupSyncPacket(ServerGroup group, boolean removal) {
        this.group = group;
        this.removal = removal;
    }

    @Override
    public void handle(@NotNull ServerClient client) {}

    @Override
    public void encodePayload(PacketData packetData) {
        packetData.writeAll(group, removal);
    }

    @Override
    public void decodePayload(PacketData packetData) {}

    public static ServerGroupSyncPacket create(ServerGroup group, boolean removal) {
        return new ServerGroupSyncPacket(group, removal);
    }
}
